app/models/concerns/devise_setup.rb

```diff
@@ -0,0 +1,29 @@
+module DeviseSetup
+  require 'hours'
+
+  extend ActiveSupport::Concern
+
+  included do
+    if ActiveRecord::Base.connection.table_exists? 'users'
+      if User.count == 0
+        devise :database_authenticatable,
+               :recoverable,
+               :rememberable,
+               :trackable,
+               :validatable,
+               :confirmable,
+               :registerable,
+               :invitable
+      else
+        devise :database_authenticatable,
+               :recoverable,
+               :rememberable,
+               :trackable,
+               :validatable,
+               :confirmable,
+               :invitable
+      end
```

Comment:
I don't really like this approach tbh, but if we go for this we could remove the duplication:
```
devise_options = [:database_authenticatable, :recoverable, :rememberable, :trackable, :validatable, :confirmable, :registerable, :invitable]
devise_options << :registerable if User.count == 0
devise *devise_options
```